src/lib/context.py

```diff
@@ -90,6 +116,21 @@ def log(self, *args):
 
         print(f"{timestamp_utc_iso} {context_section} : {message}")
 
+    def __check_if_message_exceeded_limit(self, message: str):
+        log_calls_performed = self.throttled_log_call_count.get(message, 0)
```

Comment:
I'm afraid of memory usage in case of very high number of similar messages but with small difference (e.g. containing timestamp).
log_calls_performed map should have expiration policy - be somehow  periodically cleared of old entries

Reply:
LoggingContexts are created per single polling run and the map is a field of this object. So map will be recreated for every single polling.
For LoggingContext created at the start of the container - it's used only on start, so there is no need to throttle logs for it.

Reply:
I was wondering about some type of list of timestamps for recurring logs, but I feel like it's overkill. Especially, that we have never had problems with lots of logs for GCP.